trace-normalization/src/normalizer.rs

```diff
@@ -103,11 +114,72 @@ pub(crate) fn is_valid_status_code(sc: &str) -> bool {
     false
 }
 
+/// normalize_trace takes a trace and
+/// * returns an error if there is a trace ID discrepancy between 2 spans
+/// * returns an error if at least one span cannot be normalized
+pub fn normalize_trace(trace: &mut [pb::Span]) -> anyhow::Result<()> {
+    let first_trace_id = match trace.get(0) {
+        Some(first_span) => first_span.trace_id,
+        None => anyhow::bail!("Normalize Trace Error: Trace is empty"),
+    };
+
+    for span in trace {
```

Comment:
maybe you can simplify this a bit further using the `?` operator
something like
```
for span in trace {
  if span.trace_id != first_trace_id {
    return Err(format!("Normalize Trace Error: Trace has foreign span: {:?}", span));
  }
  normalize_span(span)?;
}
Ok(())
```

Reply:
changed!